api/v1alpha4/metal3machinetemplate_types.go

```diff
@@ -23,6 +23,12 @@ import (
 // Metal3MachineTemplateSpec defines the desired state of Metal3MachineTemplate
 type Metal3MachineTemplateSpec struct {
 	Template Metal3MachineTemplateResource `json:"template"`
+
+	// When set to True, CAPM3 Machine controller will
+	// pick the same pool of BMHs' that was released during the upgrade operation.
+	// +kubebuilder:default=false
+	// +optional
+	NodeReuse bool `json:"nodeReuse"`
```

Comment:
Added in [here](https://github.com/metal3-io/cluster-api-provider-metal3/blob/master/docs/api.md#metal3machinetemplate)

Reply:
What was the main use case that led `NodeReuse` to added on the metal3machinetemplate and not on metal3machine?

Reply:
AFAIK, it is mainly because nodeReuse feature will be used in upgrade/remediation scenarios, and in that case, it is better to land the field in the higher-level object (in M3MTemplate), the reason being the user can manage the field as it may not be always desirable to use this feature, whereas it is impossible to manage metal3machines since they are controlled by CAPM3 machine controller. Additionally, proposal is already merged following the same design in this PR.